### ISSUE TITLE
Clearance violations caused by the multithreaded route optimizer (use "-mt 1" to disable it)

### ISSUE THREAD

**Issue by ProxyPlayerHD** [freerouting/freerouting#155]
* What version of freerouting are you using?
v1.6.2 (build 2022-07-13)

* On what platform did you run it?
Windows 10

* What type of file are you working with?
Spectra DNS exported from KiCad

* Did you use any command line arguments?
"-l en"

* Can you post a screenshot of the problem?
<img width="1250" alt="freerouting_xjfaaj0YM8" src="https://user-images.githubusercontent.com/13576843/201993956-bd1952bc-238e-4608-a12e-468fd47b8868.png">

* Please list the detailed steps to reproduce the problem
Click "Autorouter", wait for it to finish

* What would you expect freerouting to do in this situation?
not this

* Is fixing this issue critical for your work?
nah i'm just a hobbyist, but having the autorouter not work kinda defeats the purpose of it...

DNS and Log are in this Zip File: 
[FR Stuff.zip](https://github.com/freerouting/freerouting/files/10015149/FR.Stuff.zip)

hope this is enough info to find out what is happening

**Comment by DLehenbauer** [comment]
@ProxyPlayerHD - I'm curious if you tried '-mt 1' to disable multithreading?

Anecdotally, I've found that this problem is exacerbated on fast machines with many cores.

**Comment by DomoticX** [comment]
I can confirm this issue, using version 1.6.2 and 1.6.5....when using all cores (32 on R9-7950X), some tracks just go trough the pads, using the '-mt 1' command solves this problem for now.

**Comment by ProxyPlayerHD** [comment]
i could've sworn i already commented... hmm.
anyways yes, using `-mt 1` avoids the issue. of course that isn't a solution, but it works for now.

**Comment by andrasfuchs** [comment]
Thank you guys for the reports, it would be important to fix this in the next release.

I agree that if I can't allocate the time for the bugfix in the next week or so, it would be better to have a new release with multithreading disabled by default.

**Comment by andrasfuchs** [comment]
Alright, so I spent a few hours on this issue, and although I cleaned up the multithreading task scheduler code, I don't think it makes sense to use it. The multi-threader tries to optimize multiple items on the board at the same time, but if they are related it can't guarantee the clearances. I think it is much better to keep it single threaded even if this means that the optimization process takes a little longer.

I will keep the `-mt` option for now just in case we want to run experiments, but its default value will be 1, so the route optimizer will be single threaded.
